### PR TITLE
feat: create notification on user catalog issue

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding notification when the catalog is malformed or wrong version is used.

### Screenshot / video of UI

https://github.com/user-attachments/assets/15004af9-0ceb-4d81-99e9-858c95f39013

**using old catalog version**
image

**malformed catalog**
image

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1394

### How to test this PR?
- [x] unit tests has been provided

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Migration guide
+
+## ApplicationCatalog
+
+Before **Podman AI Lab** `v1.2.0` the [user-catalog](./PACKAGING-GUIDE.md#applicationcatalog) were not versioned. 
+Started from `v1.2.0` all user-catalog require to have a `version` property.
```
Comment (lstocchi) [inline MIGRATION.md:6]:
```suggestion
Starting from `v1.2.0` all user-catalog require to have a `version` property.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The list of catalog versions can be found in [packages/backend/src/utils/catalogUtils.ts](https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/packages/backend/src/utils/catalogUtils.ts)
+
+The catalog has its own version number, as we may not require to update it with every update. It will follow semantic versioning convention.
+
+## `None` to Catalog `1.0`
+
+None represents any catalog version prior to the first versioning. 
+
+Version `1.0` of the catalog add an important property to models `backend`, defining the type of framework required by the model to run (E.g. LLamaCPP, WhisperCPP).
```
Comment (lstocchi) [inline MIGRATION.md:16]:
```suggestion
Version `1.0` of the catalog adds an important property to models `backend`, defining the type of framework required by the model to run (E.g. LLamaCPP, WhisperCPP).
```

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Before **Podman AI Lab** `v1.2.0` the [user-catalog](./PACKAGING-GUIDE.md#applicationcatalog) were not versioned. 
+Started from `v1.2.0` all user-catalog require to have a `version` property.
+
+The list of catalog versions can be found in [packages/backend/src/utils/catalogUtils.ts](https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/packages/backend/src/utils/catalogUtils.ts)
+
+The catalog has its own version number, as we may not require to update it with every update. It will follow semantic versioning convention.
+
+## `None` to Catalog `1.0`
+
+None represents any catalog version prior to the first versioning.
```
Comment (lstocchi) [inline MIGRATION.md:14]:
```suggestion
`None` represents any catalog version prior to the first versioning. 
```

**Review by lstocchi — APPROVED** [review]
LGTM, thanks for adding the notification 🥳

**Comment by axel7083** [comment]
> LGTM, thanks for adding the notification 🥳

Thanks for the reading proof, I'm very bad at writing documentation!

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Migration guide
+
+## ApplicationCatalog
+
+Before **Podman AI Lab** `v1.2.0` the [user-catalog](./PACKAGING-GUIDE.md#applicationcatalog) were not versioned.
```
Comment (jeffmaury) [inline MIGRATION.md:5]:
```suggestion
Before **Podman AI Lab** `v1.2.0` the [user-catalog](./PACKAGING-GUIDE.md#applicationcatalog) was not versioned. 
```

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Migration guide
+
+## ApplicationCatalog
+
+Before **Podman AI Lab** `v1.2.0` the [user-catalog](./PACKAGING-GUIDE.md#applicationcatalog) were not versioned. 
+Starting from `v1.2.0` all user-catalog require to have a `version` property.
```
Comment (jeffmaury) [inline MIGRATION.md:6]:
```suggestion
Starting from `v1.2.0` the user-catalog require to have a `version` property.
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**